### DISCUSSION TITLE
How to force regenerating localised HTML pages?

### DISCUSSION THREAD

**Discussion by nclm** [WeblateOrg/weblate#12334] [Q&A]
Hello,
I have a small HTML page hosted on Codeberg Pages which uses the Weblate provided by Codeberg to let people make translations of the page. It works quite well!
However, when I make changes on the HTML page that don’t involve changing strings, I cannot figure out how to propagate these changes to the localised HTML pages generated by Weblate.
I cannot click Push in Repository maintenance because the translations have not been changed.
Is there a way to force regenerating the localised pages, or anything else that would have this effect? Thanks!

**Comment by nclm** [comment]
I discovered the option `Force writing all translations to the Weblate repository` which _seems_ to do what I want. Since it is in the “Danger zone” though, I would like to know what I actually risk by using this option? What kind of data loss is possible by clicking `Synchronize`?

**Reply by nijel** [reply]
It should be safe to perform, but the intention was that this is typically not an operation user will need.

**Reply by nclm** [reply]
Thanks. Is there another way to regenerate the localised HTML files? Because in our case, we need to perform it at almost every change so that the changes propagate to all the localisations. I will use this “Force writing” function for now, but if there is anything more relevant for the case, would love to know about it.

**Reply by nijel** [reply]
Cleanup add-on seems to do the trick (see https://github.com/WeblateOrg/weblate/discussions/12123#discussioncomment-10347128). Maybe we should consider enabling it automatically for HTML.

**Reply by nijel** [reply]
...and I've just done that in https://github.com/WeblateOrg/weblate/pull/12337
